pkg/config/leaderelection/leaderelection.go

```diff
@@ -85,13 +85,21 @@ func LeaderElectionDefaulting(config configv1.LeaderElection, defaultNamespace,
 	ret := *(&config).DeepCopy()
 
 	if ret.LeaseDuration.Duration == 0 {
-		ret.LeaseDuration.Duration = 60 * time.Second
+		// account for 20% jitter of the RetryPeriod
+		ret.LeaseDuration.Duration = 99 * time.Second
 	}
+	// these total 30s and then 20% jitter on RetryPeriod.  This gives us two properties
+	//  1. We can span a 60s kube-apiserver communication interruption (30+jitter1 + 30+jitter2) is always at least 60s.
+	//  2. We can always retry at least three times inside of our leaseDuration even in cases of long jitter and long
+	//     waits (black hole) that causes us to bump the RenewDeadline.  In cases of a short renew failure, we may retry
+	//     more often.
 	if ret.RenewDeadline.Duration == 0 {
-		ret.RenewDeadline.Duration = 35 * time.Second
+		// we choose a very long RenewDeadline so that case of very slow etcd are exacerbated by lost leases and
+		// cascading retries.
+		ret.RenewDeadline.Duration = 17 * time.Second
 	}
 	if ret.RetryPeriod.Duration == 0 {
-		ret.RetryPeriod.Duration = 10 * time.Second
+		ret.RetryPeriod.Duration = 13 * time.Second
```

Comment:
Note that a 13s retry period is very short.  Most components would be ok with 30s retry period, 60s renew deadline, and 90s lease.  For a core critical controller 13s is reasonable.